build.sbt

```diff
@@ -1,90 +1,209 @@
+    "-Xlint:type-parameter-shadow", // a local type parameter shadows a type already in scope
+    "-Xlint:poly-implicit-overload", // parameterized overloaded implicit methods are not visible as view bounds
+    "-Xlint:option-implicit", // Option.apply used implicit view
+    "-Xlint:delayedinit-select", // Selecting member of DelayedInit
+    "-Xlint:package-object-classes" // Class or object defined in package object
+
+  ),
+
+  // ScalaDoc settings
+  autoAPIMappings := true,
+  scalacOptions in ThisBuild ++= Seq(
+    // Note, this is used by the doc-source-url feature to determine the
+    // relative path of a given source file. If it's not a prefix of a the
+    // absolute path of the source file, the absolute path of that file
+    // will be put into the FILE_SOURCE variable, which is
+    // definitely not what we want.
+    "-sourcepath",
+    file(".").getAbsolutePath.replaceAll("[.]$", "")
+  ),
+  parallelExecution in Test             := false,
+  parallelExecution in IntegrationTest  := false,
+  parallelExecution in ThisBuild        := false,
+  testForkedParallel in Test            := false,
+  testForkedParallel in IntegrationTest := false,
+  testForkedParallel in ThisBuild       := false,
+  concurrentRestrictions in Global += Tags.limit(Tags.Test, 1),
+  logBuffered in Test            := false,
+  logBuffered in IntegrationTest := false,
+  // https://github.com/sbt/sbt/issues/2654
+  incOptions := incOptions.value.withLogRecompileOnMacro(false),
+
+  // todo add sonatype
+  // -- Settings meant for deployment on oss.sonatype.org
+  /*sonatypeProfileName := organization.value,
+  credentials += Credentials(
+    "Sonatype Nexus Repository Manager",
+    "oss.sonatype.org",
+    sys.env.getOrElse("SONATYPE_USER", ""),
+    sys.env.getOrElse("SONATYPE_PASS", "")
+  ),*/
+
+  publishMavenStyle := true,
+  publishTo := Some(
+    if (isSnapshot.value)
+      Opts.resolver.sonatypeSnapshots
+    else
+      Opts.resolver.sonatypeStaging
+  ),
+  isSnapshot              := version.value endsWith "SNAPSHOT",
+  publishArtifact in Test := false,
+  pomIncludeRepository    := { _ => false }, // removes optional dependencies
+  // For evicting Scoverage out of the generated POM
+  // See: https://github.com/scoverage/sbt-scoverage/issues/153
+  pomPostProcess := { (node: xml.Node) =>
+    new RuleTransformer(new RewriteRule {
+      override def transform(node: xml.Node): Seq[xml.Node] = node match {
+        case e: Elem
+            if e.label == "dependency" && e.child.exists(child =>
+              child.label == "groupId" && child.text == "org.scoverage") =>
+          Nil
+        case _ => Seq(node)
+      }
+    }).transform(node).head
+  },
+  licenses      := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt")),
+  //homepage := Some(url("https://monix.io")), //todo homepage settings
+  headerLicense := Some(HeaderLicense.Custom("""|Copyright (c) 2014-2020 by The Monix Project Developers.
```

Comment:
I have this preference, nay obsession, for indenting to not look like this with things hanging from the far right. It looks like a branch of some sort 🙂

Just my preference, no need to do anything about it.